bee-protocol/src/workers/peer/manager_res.rs

```diff
@@ -101,13 +101,18 @@ impl PeerManager {
         self.inner.read().peers.is_empty()
     }
 
-    // TODO find a way to only return a ref to the peer.
-    pub fn get(&self, id: &PeerId) -> Option<impl std::ops::Deref<Target = PeerTuple> + '_> {
-        RwLockReadGuard::try_map(self.inner.read(), |map| map.get(id)).ok()
+    pub fn get_map<T>(&self, id: &PeerId, f: impl FnOnce(&PeerTuple) -> T) -> Option<T> {
+        let guard = self.inner.read();
+        let output = guard.get(id).map(f);
+        drop(guard);
+        output
```

Comment:
I guess the body is so complicated because of lifetime issues. Does the lifetime of `guard` has to be extended due to it being dropped after `get`. I honestly hate this issue if that's the case here. Otherwise we should really be able to write `self.inner.read().get(id).map(f)` here, right?